subiquity/server/controllers/install.py

```diff
@@ -130,12 +162,92 @@ async def configure_apt(self, *, context):
         configurer = await mirror.wait_config()
         return await configurer.configure_for_install(context)
 
+    async def run_curtin_install_step(
```

Comment:
I think it would be good to have each curtin step go through the context machinery, possibly by adding a @with_context decorator here?

Reply:
Good point, added a `with_context` to this function:

```
+    @with_context(
+        description="executing curtin install {step.name} step")
```